model/src/pyrenew/process/rtweeklydiff.py

```diff
@@ -0,0 +1,222 @@
+        prior: any,
+        autoreg: any,
+        sigma_r: any,
+    ) -> None:
+        """
+        Validate the input parameters.
+
+        Parameters
+        ----------
+        n_obs : int
+            Number of observations.
+        weekday_data_starts : int
+            Weekday data starts.
+        prior : any
+            Log Rt prior for the first two observations.
+        autoreg : any
+            Autoregressive parameter.
+        sigma_r : any
+            Standard deviation of the noise.
+
+        Returns
+        -------
+        None
+        """
+
+        # Nweeks should be a positive integer
+        assert n_obs > 0, f"n_obs should be a positive integer. It is {n_obs}."
+
+        # Weekday data starts should be a positive integer between 0 and 6
+        assert 0 <= weekday_data_starts <= 6, (
+            "weekday_data_starts should be a positive integer between 0 and 6."
+            + f"It is {weekday_data_starts}."
+        )
+
+        _assert_sample_and_rtype(prior)
+        _assert_sample_and_rtype(autoreg)
+        _assert_sample_and_rtype(sigma_r)
+
+        return None
+
+    @staticmethod
+    def autoreg_process(
+        dat: ArrayLike, sigma: float
+    ) -> tuple[ArrayLike, float]:
+        """
+        Scan function for the autoregressive process.
+
+        Parameters
+        ----------
+        dat : ArrayLike
+            Data array with three elements: log_rt0, log_rt1, and b.
+        sigma : float
+            Standard deviation of the noise.
+
+        Returns
+        -------
+        tuple
+        """
+
+        log_rt0, log_rt1, b = dat
+
+        next_log_rt = log_rt1 + b * (log_rt1 - log_rt0) + sigma
+
+        return jnp.hstack([log_rt1, next_log_rt, b]), next_log_rt
+
+    def sample(
+        self,
+        duration: int | None = None,
+        **kwargs,
+    ) -> RtWeeklyDiffSample:
+        """
+        Samples the weekly Rt with autoregressive difference.
+
+        Parameters
+        ----------
+        duration : int, optional
+            Duration of the sequence. Defaults to None, in which case it is
+            set to the number of observations (`self.n_obs`).
+        **kwargs : dict, optional
+            Additional keyword arguments passed through to internal sample()
+            calls, should there be any.
+
+        Returns
+        -------
+        RtWeeklyDiffSample
+            Named tuple with "rt".
+        """
+
+        # Initial sample
+        log_rt_prior = self.log_rt_prior.sample(**kwargs)[0]
+        b = self.autoreg.sample(**kwargs)[0]
+        s_r = self.sigma_r.sample(**kwargs)[0]
+
+        # Sample noise
+        noise = npro.sample(
+            self.site_name + "_error",
+            dist.Normal(0, s_r),
+            sample_shape=(self.n_weeks,),
+        )
+
+        # Running the process
+        _, log_rt = lax.scan(
+            f=self.autoreg_process,
+            init=jnp.hstack([log_rt_prior, b]),
+            xs=noise,
+        )
```

Comment:
I'm guessing `FirstDifferenceAR` would replace this bit? But I'm unsure how it would go. LMK if you have any time to discuss, @dylanhmorris.

Reply:
Discussed sync, but you'd use it for lines 194-206

Reply:
Added in 0bf05cf (and fixed later on)